test/integration/controllermanager/managedseedset/managedseedset_test.go

```diff
@@ -0,0 +1,383 @@
+			managedSeedList.Items[0].Status = seedmanagementv1alpha1.ManagedSeedStatus{
+				ObservedGeneration: managedSeedList.Items[0].GetGeneration(),
+				Conditions: []gardencorev1beta1.Condition{
+					{Type: seedmanagementv1alpha1.ManagedSeedSeedRegistered, Status: gardencorev1beta1.ConditionTrue},
+				},
+			}
+			Expect(testClient.Status().Patch(ctx, &managedSeedList.Items[0], patch)).To(Succeed())
+
+			By("Create Seed manually as ManagedSeed controller is not running in the test")
+			seed.Name = managedSeedList.Items[0].Name
+			Expect(testClient.Create(ctx, seed)).To(Succeed())
+			Eventually(func(g Gomega) {
+				g.Expect(testClient.List(ctx, seedList, client.MatchingLabelsSelector{Selector: selector})).To(Succeed())
+				g.Expect(seedList.Items).To(HaveLen(1))
+			}).Should(Succeed())
+
+			By("Mark the Seed as Ready")
+			Expect(testClient.Get(ctx, client.ObjectKeyFromObject(seed), seed)).To(Succeed())
+			patch = client.MergeFrom(seed.DeepCopy())
+			seed.Status = gardencorev1beta1.SeedStatus{
+				ObservedGeneration: seed.GetGeneration(),
+				Conditions: []gardencorev1beta1.Condition{
+					{Type: gardencorev1beta1.SeedGardenletReady, Status: gardencorev1beta1.ConditionTrue},
+					{Type: gardencorev1beta1.SeedBootstrapped, Status: gardencorev1beta1.ConditionTrue},
+					{Type: gardencorev1beta1.SeedSystemComponentsHealthy, Status: gardencorev1beta1.ConditionTrue},
+					{Type: gardencorev1beta1.SeedBackupBucketsReady, Status: gardencorev1beta1.ConditionTrue},
+				},
+			}
+			Expect(testClient.Status().Patch(ctx, seed, patch)).To(Succeed())
+
+			DeferCleanup(func() {
+				By("Delete Seed")
+				Expect(testClient.Delete(ctx, seed)).To(Or(Succeed(), BeNotFoundError()))
+				Eventually(func() error {
+					return testClient.Get(ctx, client.ObjectKeyFromObject(seed), seed)
+				}).Should(BeNotFoundError())
+			})
+
+			Eventually(func(g Gomega) {
+				g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(managedSeedSet), managedSeedSet)).To(Succeed())
+				g.Expect(managedSeedSet.Status.ReadyReplicas).To(Equal(int32(1)))
+			}).Should(Succeed())
+		})
+	})
+
+	Context("scaleOut", func() {
+		It("should create new replica and update the status.Replicas field because spec.Replica value is increased", func() {
+			Eventually(func(g Gomega) {
+				g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(managedSeedSet), managedSeedSet)).To(Succeed())
+				g.Expect(managedSeedSet.Status.Replicas).To(Equal(int32(1)))
+			}).Should(Succeed())
+
+			By("Update the Replicas to 2")
+			patch := client.MergeFrom(managedSeedSet.DeepCopy())
+			managedSeedSet.Spec.Replicas = pointer.Int32(2)
+			Expect(testClient.Patch(ctx, managedSeedSet, patch)).To(Succeed())
+
+			By("Mark the pending replica to nil, to enable controller to pick the next replica")
+			patch = client.MergeFrom(managedSeedSet.DeepCopy())
+			managedSeedSet.Status.PendingReplica = nil
+			Expect(testClient.Status().Patch(ctx, managedSeedSet, patch)).To(Succeed())
```

Comment:
Why is this needed?

Reply:
The controller moves to next only when there is no pending replica and the earlier replica is ready. In order to make the first replica ready , it needs to satisfy all the conditions for the readiness i.e Shoot is healthy and successfully created, ManagedSeed has SeedRegistered condition, and Seed is ready. As it is already done in earlier test. Hence I directly set the `pendingReplica` to be nil, to show that once the pending replica is nil the controller move on to the next replica and updates the counter for replicas in status field.